src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+//#define LAUNCH_DEBUGGER
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.Extensions.Configuration.SourceGeneration
+{
+    /// <summary>
+    /// Generates source code to optimize binding with ConfigurationBinder.
+    /// </summary>
+    [Generator]
+    public sealed partial class ConfigurationBinderSourceGenerator : IIncrementalGenerator
+    {
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            IncrementalValuesProvider<InvocationExpressionSyntax> inputCalls = context.SyntaxProvider.CreateSyntaxProvider(
+                (node, _) => Parser.IsInputCall(node),
+                (syntaxContext, _) => (InvocationExpressionSyntax)syntaxContext.Node);
+
+            IncrementalValueProvider<(Compilation, ImmutableArray<InvocationExpressionSyntax>)> compilationAndClasses =
+                context.CompilationProvider.Combine(inputCalls.Collect());
```

Comment:
We shouldn't be doing this:

https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md#authoring-a-cache-friendly-generator

Reply:
I appreciate the perf pointers, this catches me up with the latest thinking in the space.